src/components/Favorite/Favorite.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { addFavorite } from '../../actions/favorite'
+
+export class Favorite extends React.Component {
+
+    favorite = () => {
+        const { articleId, hideOptions, dispatch } = this.props
+        hideOptions() // hide more options popup
+        dispatch(addFavorite(articleId))
+    }
+
+    render() {
+        const { setMsg, message } = this.props
```

Comment:
why not `setMessage` instead of `setMsg`
```suggestion
        const { setMessage, message } = this.props
```